### PR TITLE
rgw_sal_motr, motr_gc: [CORTX-32689] Implement MotrGC::enqueue()

### PR DESCRIPTION
**Pull request by sumedhak27**
* Push `{0_ObjTag: motr_gc_obj_info}` and `{1_ExpiryTime: motr_gc_obj_info}`
  entry to the gc queues, i.e. motr dix.
* Call gc->enqueue() method on simple object delete request without
  actually deleting the obj.

Signed-off-by: Sumedh Anantrao Kulkarni <sumedh.a.kulkarni@seagate.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/motr/gc/gc.cc [hunk 1]
```diff
@@ -145,33 +150,58 @@ bool MotrGC::going_down() {
...
+  bool index_locked = true;
   // attempt to lock GC starting with passed in index
-  for (uint32_t ind = 0; ind < max_indices; ind++)
-  {
+  for (uint32_t ind = 1; ind < max_indices; ind++) {
     new_index = (ind + rand_ind) % max_indices;
     // try locking index
     // on sucess mark rc as 0
+    if (index_locked)
+      break;
```
Comment (sumedhak27) [inline src/rgw/motr/gc/gc.cc:204]: index locked will be initialized as `false` and set to `true` on the successful acquisition of the lock, till then processing any gc queue that is requested.

File: src/rgw/motr/gc/gc.cc [hunk 2]
```diff
@@ -145,33 +150,58 @@ bool MotrGC::going_down() {
...
+  // push {0_ObjTag: motr_gc_obj_info} to the gc queue.📥
+  rc = store->do_idx_op_by_name(index_names[enqueue_index],
+                                M0_IC_PUT, key1, bl);
+  if (rc < 0)
+    return rc;
+  
+  // push {1_ExpiryTime: motr_gc_obj_info} to the gc queue.📥
+  rc = store->do_idx_op_by_name(index_names[enqueue_index],
+                                M0_IC_PUT, key2, bl);
+  if (rc < 0)
```
Comment (sachinpunadikar) [inline src/rgw/motr/gc/gc.cc:171]: here remove the added 0_ObjTag entry from index before returning error

File: src/rgw/rgw_sal_motr.cc [hunk 3]
```diff
@@ -2219,7 +2220,18 @@ int MotrObject::remove_mobj_and_index_entry(
...
+        std::string tag = PRIx64 + std::to_string(this->meta.oid.u_hi) + ":" +
+                          PRIx64 + std::to_string(this->meta.oid.u_lo);
+        std::string obj_fqdn = bucket_name + delete_key;
+        ::Meta *mobj = reinterpret_cast<::Meta*>(&this->meta);
+        motr_gc_obj_info obj(tag, obj_fqdn, *mobj, std::time(nullptr),
+                             ent.meta.size, size_rounded, false, "");
+        rc = store->get_gc()->enqueue(obj);
+        pushed_to_gc = (rc == 0);   
+      }
+      if (! pushed_to_gc)
```
Comment (DPG17) [inline src/rgw/rgw_sal_motr.cc:2233]: if (! pushed_to_gc) evaluates to false, you can log a message indicating object (oid, pv id, layout id) would be deleted by Motr GC. Would be useful by QA for validation.

Comment (sumedhak27) [reply src/rgw/rgw_sal_motr.cc:2237]: addressed. 👍

File: src/rgw/motr/gc/gc.cc (hunk 2 above)
Comment (sumedhak27) [reply src/rgw/motr/gc/gc.cc]: addressed.

**Review by sachinpunadikar — APPROVED** [review]
LGTM

**Review by jjxsg — APPROVED** [review]

**Review by DPG17 — APPROVED** [review]

**Pull request merged**